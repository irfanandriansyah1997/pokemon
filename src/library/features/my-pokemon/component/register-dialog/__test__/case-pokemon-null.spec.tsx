import { mount } from 'enzyme';

import MyPokemonProvider from '@/library/features/my-pokemon';
import PokemonRegisterDialog from '@/library/features/my-pokemon/component/register-dialog';

jest.mock(`@/library/features/my-pokemon/hooks/my-pokemon.hooks.ts`);

describe(`Case Pokemon Null`, () => {
  beforeAll(() => {
    localStorage.setItem(`pokemon-list`, JSON.stringify([]));
  });

  afterEach(() => {
    jest.spyOn(global.Math, `random`).mockRestore();
  });

  it(`Testing Render`, async () => {
    const onSpy = jest.fn((param) => param);
    const ui = mount(
      <MyPokemonProvider>
        <PokemonRegisterDialog
          on={onSpy}
          saved={false}
          pokemon={undefined}
          show
        />
      </MyPokemonProvider>
    );

    expect(ui.html()).toMatchSnapshot();
  });
});
