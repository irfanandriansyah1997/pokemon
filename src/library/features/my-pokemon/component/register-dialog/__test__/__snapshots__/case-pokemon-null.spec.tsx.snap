// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Case Pokemon Null Testing Render 1`] = `""`;
